### PR TITLE
Feat: Add teamdctl_state_dump spec to insights_archive

### PR DESCRIPTION
**Pull request by TZ3070**
Signed-off-by: Xinting Li <xintli@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

teamdctl_state_dump spec only exists in sos-archive.py currently, but some insights-plugin rules also use the related parser.

Add teamdctl_state_dump spec to default.py.

### CODE REVIEW

File: insights/specs/default.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ class DefaultSpecs(Specs):
     ethtool_c = foreach_execute(ethernet.interfaces, "/sbin/ethtool -c %s")
     ethtool_g = foreach_execute(ethernet.interfaces, "/sbin/ethtool -g %s")
     ethtool_i = foreach_execute(ethernet.interfaces, "/sbin/ethtool -i %s")
-    ethtool_k = foreach_execute(ethernet.interfaces, "/sbin/ethtool -k %s")
+    ethtool_k = foreach_execute(ethernet.interfaces, "/bin/ethtool -k %s")
```
Comment (xiangce) [inline insights/specs/default.py:193]: @TZ3070 - Is this change required?

Comment (TZ3070) [reply insights/specs/default.py]: No, it is a mistake.

Comment (xiangce) [reply insights/specs/default.py]: @TZ3070 - This update looks good to me. But as it's also a kind of SPEC change to the core collection, please raise an issue (not MR) to the insights-core-assets repo for this change.

Comment (psachin) [reply insights/specs/default.py]: @TZ3070 Can you please verify is the path is `/bin/teamdctl`? On my rhel8.5 system it shows `/usr/bin/teamdctl`

Comment (TZ3070) [reply insights/specs/default.py]: @psachin 
After checking, The path is '/usr/bin/teamdctl'. '/bin/teamdctl' is symbolic link.

**Review by psachin — APPROVED** [review]

**Review by xiangce — CHANGES_REQUESTED** [review]
@TZ3070 - please check the comments to the Spec change and update this PR accordingly. Thanks.

File: insights/parsers/nmcli.py [hunk 2]
```diff
@@ -176,10 +176,20 @@ class NmcliConnShow(CommandParser):
         data (list): list of connections wrapped in dictionaries
 
     """
+
     def parse_content(self, content):
-        try:
-            self.data = parse_delimited_table(content, heading_ignore=["NAME", "UUID", "TYPE", "DEVICE"])
-        except:
+        self.data = []
```
Comment (xiangce) [inline insights/parsers/nmcli.py:181]: @TZ3070 - better fix this issue #3461 in an individual PR but not mix them together.

**Comment by TZ3070** [comment]
When using the NmcliConnShow parser in datasource 'team', I found the parser cannot handle the NAME containing space in the output. 

Issue: #3461
Enhance PR: https://github.com/RedHatInsights/insights-core/pull/3465

File: insights/specs/datasources/team.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+"""
```
Comment (xiangce) [inline insights/specs/datasources/team.py:1]: @TZ3070 - I believe your method works as expected, however, we generally implement the dependencies via `component` where there are existing parsers can be leveraged.

Please check the following examples:
https://github.com/RedHatInsights/insights-core/blob/b12686290b8962b2f504efebef3ebb86cf3d733f/insights/specs/default.py#L75

https://github.com/RedHatInsights/insights-core/blob/b12686290b8962b2f504efebef3ebb86cf3d733f/insights/specs/default.py#L109

would you please change this datasource to `compoent`?

File: insights/specs/datasources/ethernet.py [hunk 4]
```diff
@@ -55,3 +56,32 @@ def interfaces(broker):
...
+
+    Raises:
+        SkipComponent: When there is not any content.
+    """
+
+    content = broker[NmcliConnShow].data
+    if content:
+        team_device_list = []
+        for x in content:
+            if 'team' in x['TYPE']:
```
Comment (xiangce) [inline insights/specs/datasources/ethernet.py:81]: Would you please double confirm if the `disconnected` team interfaces good candidates to this parser?

File: insights/specs/datasources/ethernet.py [hunk 5]
```diff
@@ -55,3 +56,32 @@ def interfaces(broker):
...
+
+    Sample data returned::
+
+        ['team0', 'team1']
+
+    Returns:
+        list: List of the team device.
+
+    Raises:
+        SkipComponent: When there is not any content.
```
Comment (xiangce) [inline insights/specs/datasources/ethernet.py:74]: This would be `When there is not any team interfaces.`

File: insights/specs/datasources/ethernet.py [hunk 6]
```diff
@@ -55,3 +56,32 @@ def interfaces(broker):
             return sorted(ifaces)
 
     raise SkipComponent
+
+
+@datasource(NmcliConnShow, HostContext)
+def team_device(broker):
```
Comment (xiangce) [inline insights/specs/datasources/ethernet.py:62]: Better rename it as `team_interfaces`?

File: insights/specs/datasources/ethernet.py (hunk 4 above)
Comment (TZ3070) [reply insights/specs/datasources/ethernet.py]: Updated

**Review by xiangce — APPROVED** [review]
good to me.

**Review by psachin — APPROVED** [review]
